### PR TITLE
task 4

### PR DESCRIPTION
**Pull request by andreikasmykou**
1 - Task 4.1 is implemented
3 - Task 4.2 is implemented lambda links are provided and returns data
4 - Task 4.3 is implemented lambda links are provided and products are stored in DB (call TASK 4.2 to see the product)
5 - Frontend application is integrated with product service (/products API) and products from product-service are represented on Frontend. Link to a working Front-End application is provided for cross-check reviewer.

+1 - POST/products lambda function returns error 400 status code if product data is invalid
+1 - All lambdas return error 500 status code on any error (DB connection, any unhandled error in code)
+1 - All lambdas do console.log for each incoming requests and their arguments
+1 - Transaction based creation of the product (in case stock creation is failed then related to this stock product is not created and not ready to be used by the end user and vice versa) 

Frontend app link - https://d17q5m0kik9x1l.cloudfront.net/ 
Gateway API - https://94ynxyaj90.execute-api.eu-west-1.amazonaws.com/dev/products/

### CODE REVIEW

**Comment by dartwlad** [comment]
https://d17q5m0kik9x1l.cloudfront.net/admin/product-form/
please integrate this form with your create lambda.
image
Also please fix the product Shema, should be id, title, count, price, description

**Comment by andreikasmykou** [comment]
@dartwlad Please clarify your requirements and link them to task_4 description

**Comment by dartwlad** [comment]
*FE: One product model as a result of BE models join(product and it's stock) *
Please go to tasks and search for the current string in the task.
Fixing of this form https://d17q5m0kik9x1l.cloudfront.net/admin/product-form/ will help in future tasks and now in testing, just my recommendation.

**Comment by andreikasmykou** [comment]
Thank you for your comment. Will implement it in the next task

File: product-service/src/services/productService.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    return product;
+  };
+
+export async function createProductInDB(product: Product): Promise<Product> {
+  validateProductSchema(product);
+
+  const { title, description, price, count } = product;
+
+  const productQuery =
+    `insert into products (title, description, price) values ('${title}', '${description}', '${price.toString()}') returning *`;
```
Comment (Ilya-Valasiuk) [inline product-service/src/services/productService.ts:29]: this is anti-pattern. Better to use Parameterized query. You can find examples here -  https://node-postgres.com/features/queries **Parameterized query** section.

**Pull request merged**